src/SFA.DAS.EmployerApprenticeshipsService.Web.UnitTests/Validators/ApprenticeshipCreateOrEdit/WhenValidatingStartDate.cs

```diff
@@ -0,0 +1,70 @@
+﻿using Moq;
+using NUnit.Framework;
+using SFA.DAS.EmployerCommitments.Application.Queries.GetTrainingProgrammes;
+using SFA.DAS.EmployerCommitments.Web.ViewModels;
+
+namespace SFA.DAS.EmployerCommitments.Web.UnitTests.Validators.ApprenticeshipCreateOrEdit
+{
+    [TestFixture()]
+    public class WhenValidatingStartDate : ApprenticeshipValidationTestBase
+    {
+        [Test]
+        public void IfTheTrainingProgrammeIsValidThenShouldPassValidation()
+        {
+            //Arrange
+            ValidModel.TrainingCode = "TESTCOURSE";
+            ValidModel.StartDate = new DateTimeViewModel(1, 6, 2018);
+
+            //Act
+            var result = Validator.Validate(ValidModel);
+
+            //Assert
+            Assert.IsTrue(result.IsValid);
+        }
+
+        [Test]
+        public void IfTheTrainingProgrammeHasNotStartedThenShouldFailValidation()
+        {
+            //Arrange
+            ValidModel.TrainingCode = "TESTCOURSE";
+            ValidModel.StartDate = new DateTimeViewModel(1, 4, 2018);
+
+            //Act
+            var result = Validator.Validate(ValidModel);
+
+            //Assert
+            Assert.IsFalse(result.IsValid);
+            Assert.IsTrue(result.Errors[0].ErrorMessage.EndsWith("after 04 18"));
+        }
+
+        [Test]
+        public void IfTheTrainingProgrammeHasExpiredThenShouldFailValidation()
+        {
+            //Arrange
+            ValidModel.TrainingCode = "TESTCOURSE";
+            ValidModel.StartDate = new DateTimeViewModel(1, 8, 2018);
+
+            //Act
+            var result = Validator.Validate(ValidModel);
+
+            //Assert
+            Assert.IsFalse(result.IsValid);
+            Assert.IsTrue(result.Errors[0].ErrorMessage.EndsWith("before 08 18"));
+        }
+
+        [Test]
+        public void IfTrainingCodeIsNotSuppliedThenShouldNotCheckCourseValidity()
```

Comment:
This tests an implementation detail and could/should be removed.
(Although it could be argued that we do want to test this implementation detail in this instance, to ensure that the code under test doesn't do more work than is necessary.)

Reply:
I think it's worth keeping, because it ensures we have short-circuited the method as early as possible without incurring the cost of the api call. Perhaps the name isn't the best.